pkg/obfuscate/json_test.go

```diff
@@ -150,3 +151,41 @@ func FuzzObfuscateJSON(f *testing.F) {
 		o.obfuscate(b)
 	})
 }
+
+func TestObfuscateJSONConcurrently(t *testing.T) {
+	var (
+		cfg = &JSONConfig{
+			KeepValues: []string{"company_wallet_configuration_id"},
+		}
+		o        = newJSONObfuscator(cfg, &Obfuscator{})
+		rs       = make(chan string, 1)
+		expected = "{\"email\":\"?\",\"company_wallet_configuration_id\":1}"
+	)
+
+	// Concurrency issues were reproduced at 100 times, but we leave this at 100K
+	// to stress test the obfuscator.
+	const times = 100000
+	var wg sync.WaitGroup
+	for i := 0; i < times; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			in := "{\"email\":\"dev@datadoghq.com\",\"company_wallet_configuration_id\":1}"
+			rs <- obfuscateJSONString(in, o)
+		}()
+	}
+
+	go func() {
+		wg.Wait()
+		close(rs)
+	}()
```

Comment:
I don't think we should do this, the program can exist without waiting and closing the channel. https://stackoverflow.com/a/73994246

Reply:
@ahmed-mez I think it's a different case, as we are not sending jobs to goroutines, but collecting the results of the goroutines to check all of them returned the expected payload, without corruption.

Reply:
I think what's guaranteeing that we wait for all goroutines to finish in the current implementation is the `for` loop that waits for exactly `100000` messages (same number of go routines), which is good. But running `wg.Wait()` and `close(rs)` in a separate go routine doesn't guarantee that as the main thread can theoretically finish before, an alternative is to call `wg.Wait()` and `close(rs)` after the assertions loop, and without an extra go routine

```golang
	for i := 0; i < times; i++ {
		v := <-rs
		if v == expected {
			continue
		}
		t.Fatalf("mangled payload after obfuscation: %q", v)
	}

    wg.Wait()
    close(rs)
```

Reply:
Oh, you are right! Thanks!